.github/workflows/ci.yml

```diff
@@ -1,106 +1,105 @@
 name: CI
 
-on:
-  push:
-    branches:
-      - master
-  pull_request:
-    branches:
-      - master
+on: [push, pull_request]
 
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Ruby 2.3
-      uses: ruby/setup-ruby@master
-      with:
-        ruby-version: 2.3
-    - name: Cache Ruby dependencies
-      uses: actions/cache@v1
-      with:
-        path: ./vendor/bundle
-        key: v1-linux-2.3-${{ hashFiles('rmagick.gemspec') }}
-        restore-keys: |
-          v1-linux-2.3-${{ hashFiles('rmagick.gemspec') }}
-    - name: Build and test with Rake
-      run: |
-        bundle install --path=vendor/bundle --jobs 4 --retry 3
-        STYLE_CHECKS=true bundle exec rubocop
+      - uses: actions/checkout@v2
+      - name: Set up Ruby 2.3
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.3
+      - name: Cache Ruby dependencies
+        uses: actions/cache@v1
+        with:
+          path: ./vendor/bundle
+          key: v1-linux-2.3-${{ hashFiles('rmagick.gemspec') }}
+          restore-keys: |
+            v1-linux-2.3-${{ hashFiles('rmagick.gemspec') }}
+      - name: Build and test with Rake
+        run: |
+          bundle install --path=vendor/bundle --jobs 4 --retry 3
+          STYLE_CHECKS=true bundle exec rubocop
 
   test-linux:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby-version: [2.3, 2.4, 2.5, 2.6, 2.7]
+        ruby-version: [ 2.3, 2.4, 2.5, 2.6, 2.7, ruby-head ]
         imagemagick-version:
           - { full: 6.7.7-10, major-minor: '6.7' }
           - { full: 6.8.9-10, major-minor: '6.8' }
           - { full: 6.9.10-90, major-minor: '6.9' }
           - { full: 7.0.9-20, major-minor: '7.0' }
-
+        exclude:
+          - ruby-version: 2.6
+            imagemagick-version: { full: 6.8.9-10, major-minor: '6.8' }
```

Comment:
We currently have a lot of flaky builds on Actions. I don't think it's limited to 2.6/6.8.

Reply:
I looked again, and that build wasn't flaky.  It hung/froze on every build.

Reply:
Hmm, why would this not work on your ruby action? It seems to work fine on the main one.

Reply:
Actually, that is a new action hosted by the ruby account, with builds they create.  The build code is public, and they're fully tested.  MSFT/GitHub still has their Ruby build code private.

The build that just finished locked on Ubuntu 2.6/6.7